examples/auth/app/controllers/login.js

```diff
@@ -1,16 +1,19 @@
-import Controller from '@ember/controller';
-import { inject } from '@ember/service';
+import { action } from "@ember/object";
+import { inject } from "@ember/service";
+import Controller from "@ember/controller";
 
-export default Controller.extend({
-  session: inject(),
+export default class LoginController extends Controller {
+  @inject()
+  session;
 
-  actions: {
-    authenticate(event) {
-      event.preventDefault();
+  @action
+  authenticate(event) {
+    event.preventDefault();
 
-      this.session.authenticate('authenticator:oauth2', this.identification, this.password).catch((reason) => {
-        this.set('errorMessage', reason.error || reason);
+    this.session
+      .authenticate("authenticator:oauth2", this.identification, this.password)
+      .catch((reason) => {
+        this.set("errorMessage", reason.error || reason);
```

Comment:
This should not use `set` anymore and `errorMessage` should be a `@tracked` property.